core/src/test/scala/unit/kafka/zk/KafkaZkClientTest.scala

```diff
@@ -683,6 +688,26 @@ class KafkaZkClientTest extends ZooKeeperTestHarness {
     assertEquals(Some(brokerInfo.broker), zkClient.getBroker(1))
   }
 
+  @Test
+  def testRetryRegisterBrokerInfo(): Unit = {
+    val brokerId = 5
+    val brokerPort = 9999
+    val brokerHost = "test.host"
+    val expiredBrokerInfo = createBrokerInfo(brokerId, brokerHost, brokerPort, SecurityProtocol.PLAINTEXT)
+    expiredSessionZkClient.createTopLevelPaths()
+
+    // Register the broker, for the first time
+    expiredSessionZkClient.registerBroker(expiredBrokerInfo)
+    assertEquals(Some(expiredBrokerInfo.broker), expiredSessionZkClient.getBroker(brokerId))
+
+    // Here, the node exists already, when trying to register under a different session id,
+    // the node will be deleted and created again using the new session id.
+    expiredSessionZkClient.registerBroker(expiredBrokerInfo)
+
+    // The broker info should be the same, no error should be raised
+    assertEquals(Some(expiredBrokerInfo.broker), expiredSessionZkClient.getBroker(brokerId))
```

Comment:
Could we further verify that the czxid of the ephemeral node is different?

Reply:
Yes, thanks, good idea, doing it.